lib/search_config.rb

```diff
@@ -8,12 +8,15 @@ class SearchConfig
     spelling_index_names
     base_uri
     govuk_index_name
+    default_index_name
   ].each do |config_method|
     define_method config_method do
       elasticsearch.fetch(config_method)
     end
   end
 
+  attr_writer :elasticsearch # this is used by integration testing to override the default
```

Comment:
I would move this to an optional param in the initialiser and always set the instance variable (i.e. if nil, call `config_for("elasticsearch")`)

Otherwise there's a possibility of the config effectively changing between two reads which should never happen.

Reply:
As discussed this is a temporary (hopefully) thing that will be removed by making the config file environment aware. :)